lib/spack/spack/test/util/binary_resource.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+import shutil
+
+import pytest
+
+import spack.util.binary_resource as br
+import spack.util.executable
+from spack.paths import spack_root
+
+datadir = os.path.join(spack_root, "lib", "spack", "spack", "test", "data", "binary_resource")
+file_file = os.path.join(datadir, "file.bat")
+
+pytestmark = pytest.mark.only_windows("only runs on windows")
+
+
+@pytest.fixture
+def mock_binary_resource_root(monkeypatch, tmpdir):
+    resource_root = tmpdir.mkdir("binary-resources")
+
+    def _binary_resource_root():
+        return resource_root
+
+    monkeypatch.setattr(spack.util.binary_resource, "binary_resource_root", _binary_resource_root)
+
+
+@pytest.fixture
+def no_system(working_env):
+    """Fixture preventing system binaries from being detected"""
+    os.environ["PATH"] = ""
+
+
+@pytest.fixture
+def tar_dir():
+    return os.path.dirname(spack.util.executable.which("tar").path)
+
+
+@pytest.fixture
+def on_system(working_env):
+    """Fixture explicityly adding binary resource to PATH"""
+    env = spack.util.environment.EnvironmentModifications()
+    env.append_path("PATH", datadir)
+    env.apply_modifications()
+
+
+def test_ensure_or_acquire_no_acquire(mock_binary_resource_root, config, no_system, monkeypatch):
```

Comment:
If I understand, this and the next test mock acquisition at two different levels:

* In this case, mocking `acquire_resource` (such that issues with fetching etc. would not cause this test to fail
* In the next case, allowing `acquire_resource` to run in full

This type of coverage is rare for Spack tests. Typically I'd recommend only one (in this case, the latter); if you keep both around, I think a different name would be clearer, because this *is* calling `acquire`, and in fact we are making sure this pathway calls it. I might call it `test_try_acquire` i.e. "make sure we at least attempt to acquire a resource when we find it is not available".

Reply:
Will just remove this test.